components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationDataPublisher.java

```diff
@@ -22,9 +22,19 @@
 import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
 import org.wso2.carbon.identity.core.handler.IdentityMessageHandler;
 
-import javax.servlet.http.HttpServletRequest;
 import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
 
+/**
+ * The data publishing changed to publish data through event handling. {@link AuthenticationDataPublisher} interface not
+ * be used to register data publishing services from this release.
+ *
+ * @since 5.11.179
+ *
+ * @deprecated to use  {@link org.wso2.carbon.identity.event.event.Event} to create a event and
+ * and use {@link org.wso2.carbon.identity.event.handler.IdentityEventServiceImpl} \} for handling the event.
+ */
+@Deprecated
 public interface AuthenticationDataPublisher extends IdentityMessageHandler {
```

Comment:
Ideally we can't depricate this if we are using DataPublisherProxy

Reply:
Deprecation message was removed from this interface.